components/user/EditForm.vue

```diff
@@ -0,0 +1,294 @@
+    show: false,
+    alertRoute: '',
+    type: undefined as AlertType,
+  }
+
+  const alertState = reactive({
+    ...initialAlertState,
+  })
+
+  // Show appropriate success alert after signing up for company
+  const displaySuccessAlert = (alertRoute: string) => {
+    alertState.alertRoute = alertRoute
+    alertState.type = 'success'
+    alertState.show = true
+  }
+
+  // show appropriate error alert after signing up for company
+  const displayErrorAlert = (alertRoute: string) => {
+    alertState.alertRoute = alertRoute
+    alertState.type = 'error'
+    alertState.show = true
+  }
+
+  // Show appropriate error alert for failed API calls
+  const displayErrorAlertFromMessage = (errorMessage: string) => {
+    const content = getAlertContent(errorMessage)
+    alertState.alertRoute = content.alertRoute
+    alertState.type = content.type
+    alertState.show = content.show
+  }
+
+  const form = ref()
+
+  // save changes to user
+  const saveChanges = async () => {
+    // throw error on invalid form
+    const { valid } = await form.value.validate()
+    try {
+      if (!valid) throw new Error('Form is not valid')
+    } catch (error) {
+      displayErrorAlert('alert.error.form.invalid')
+      return
+    }
+
+    if (state.userType !== 'company') {
+      state.companyUID = null
+    }
+
+    // update user
+    await useAsyncData('update-user', async () => {
+      await Promise.all([
+        $fetch('/api/user', {
+          method: 'POST',
+          body: {
+            ...state,
+          },
+        }),
+        $fetch('/api/user', {
+          method: 'PUT',
+          body: {
+            ...state,
+          },
+        }),
+      ])
+    })
+      .then(() => {
+        displaySuccessAlert('alert.success.user.edit')
+        setTimeout(cancel, 2000)
+      })
+      .catch((error) => {
+        console.error(error)
+        displayErrorAlertFromMessage(error.message)
+      })
+  }
+
+  // navigate to edit page
+  // if user is not admin, it navigates to /user
+  const cancel = () => {
+    navigateTo(localePath('/admin/users'))
+  }
+</script>
+
+<template>
+  <!-- Header -->
+  <h4 class="text-sm-h3 text-h4 font-weight-bold text-center pt-16 pb-4">
+    {{ $t('edit.user.title') }}
+  </h4>
+
+  <!-- Alert component -->
+  <VSnackbar v-model="alertState.show">
+    {{ $t(`${alertState.alertRoute}`) }}
+
+    <template #actions>
+      <VBtn
+        :color="alertState.type"
+        variant="text"
+        @click="alertState.show = false"
+      >
+        {{ $t('alert.close_alert') }}
+      </VBtn>
+    </template>
+  </VSnackbar>
+
+  <!-- Edit Form -->
+  <VForm ref="form" @submit.prevent="saveChanges">
+    <VContainer>
+      <!-- Name -->
+      <VRow>
+        <FormTextInput
+          v-model="state.name"
+          :content="{
+            label: $t('edit.user.attributes.name'),
+          }"
+          :rules="[useRequiredInput]"
+        />
+      </VRow>
+
+      <!-- email -->
+      <VRow>

```

Comment:
Should probably verify with a regex that the mail has a valid mail format.

Reply:
fixed